README.md

```diff
@@ -83,6 +83,19 @@ OAUTH_CLIENT_ID = None
 OAUTH_CLIENT_SECRET = None
 ```
 
+## OAuth - KeyCloak
+
+To integrate with any other OAuth provider, such as KeyCloak, edit `CTFd/config.ini` and set:
+```python
+OAUTH_NAME = MyAuth
+OAUTH_CLIENT_ID = None
+OAUTH_CLIENT_SECRET = None
+OAUTH_TOKEN_ENDPOINT = https://oauth.local/realms/<yourrealm>/protocol/openid-connect/token
+OAUTH_API_ENDPOINT = https://oauth.local/realms/<yourrealm>/protocol/openid-connect/userinfo
+OAUTH_API_ENDPOINT = https://oauth.local/realms/<yourrealm>/protocol/openid-connect/userinfo
+```
+You will have to add a mapper for `id` in your client scope, and ensure `openid` is added as client scope.
```

Comment:
Any thoughts on how to handle an oauth provider where you do not have a readily available `id` value that is an integer?

Reply:
Excellent question @BrandonIngalls .
Looking at the code the this `id` is mapped to a `Users.oauth_id` and potentially to a `Teams.oauth_id`.. The requirement of it being an integer is located in the `Users` and `Teams` classes; that also triggers the DB structure integer requirement. 

Looking at the code I see no place where this `oauth_id` must be an ID. Actually it looks it's mostly used as a boolean like `{% if team.oauth_id %}`.

Summarized, unless I misunderstand the code, I think it could even be dropped from the database and transformed as a local boolean to track if the session is oauth-based or not. Thus dropping the requirement for getting an `id` from the oauth provider.